index.ts

```diff
@@ -51,15 +53,35 @@ function useLocalStorage<T>(
     }
   });
 
-  useEffect(() => {
+  const setValueInLocalStorage = useCallback((setterEntity: SetterEntity<T | undefined>) => {
     if (typeof window === "undefined") return;
 
+    const isFunction = !!setterEntity && typeof setterEntity === "function";
+    // @ts-ignore
```

Comment:
Also, I generally consider it a bit of a "code smell" if you have to ignore compiler warnings, so I'd be very hesitant to merge anything that requires ignoring warnings. Thanks!